dvc/repo/index.py

```diff
@@ -229,6 +234,9 @@ def from_file(cls, repo: "Repo", path: str) -> "Index":
             metrics={path: dvcfile.metrics} if dvcfile.metrics else {},
             plots={path: dvcfile.plots} if dvcfile.plots else {},
             params={path: dvcfile.params} if dvcfile.params else {},
+            artifacts={path: dvcfile.artifacts}
+            if hasattr(dvcfile, "artifacts")
+            else {},
```

Comment:
Is there any reason not to write this as done above?

Reply:
Sorry, I may be missing your question (please clarify if so). This is because `dvcfile` can be an instance of `SingleStageFile` and that one can't have `artifacts:` section.

Reply:
Ahh, in that case, you can just add `artifacts = {}` here:

https://github.com/iterative/dvc/blob/780dc7230174f61c1899dfbba017f0212a535fbe/dvc/dvcfile.py#L192-L194

Reply:
But is this correct? `SingleStageFile` doesn't meant to have them.

Reply:
yes, and that's what `{}` means? :smile:

Either we can specialize like you have done, or generalize. This may be temporary until we drop support for pipelines from `.dvc` file.

Reply:
Agreed, but I think better to avoid creating ambiguity for future (we can forget this discussion in future and wonder why we have `artifacts` in `SingleStageFile` and make wrong assumptions). So if you allow, I'll keep the current implementation.

Reply:
We won't forget. It has no use other than for tracking outputs, other things are deprecated. There is still a separate schema validation for `.dvc` files.

Reply:
ok, then doing it like you suggested 🙌🏻